src/main/webapp/app/entities/fda-submission/fda-submission.store.ts

```diff
@@ -1,14 +1,33 @@
 import { IFdaSubmission } from 'app/shared/model/fda-submission.model';
 import { IRootStore } from 'app/stores';
-import { action } from 'mobx';
-import axios from 'axios';
+import { action, makeObservable } from 'mobx';
+import axios, { AxiosResponse } from 'axios';
+import { notifyError } from 'app/oncokb-commons/components/util/NotificationUtils';
 import PaginationCrudStore from 'app/shared/util/pagination-crud-store';
 
 const apiUrl = 'api/fda-submissions';
 
 export class FdaSubmissionStore extends PaginationCrudStore<IFdaSubmission> {
+  lookupFdaSubmission = this.readHandler(this.lookupFdaSubmissionGen);
+
   constructor(protected rootStore: IRootStore) {
     super(rootStore, apiUrl);
+    makeObservable(this, {
+      lookupFdaSubmission: action,
+    });
+  }
+
+  *lookupFdaSubmissionGen(submissionNumber: string) {
+    const numbers = submissionNumber.split('/');
+    const url = `${apiUrl}/lookup?number=${numbers[0]}${numbers[1] ? `&supplement_number=${numbers[1]}` : ''}`;
+    try {
+      const result: AxiosResponse<IFdaSubmission> = yield axios.get(url);
```

Comment:
Why do we use `yield` here?

Reply:
`generator-jhipster-react-mobx` uses [mobx flows](https://mobx.js.org/actions.html#asynchronous-actions), which takes in a generator function, for handling async methods.